bifrost/inmemory_locker.go

```diff
@@ -0,0 +1,95 @@
+package bifrost
+
+import (
+	"fmt"
+	"sync"
+)
+
+type Lock struct {
+	lockPath string
+	uuid     string
+	reason   string
+}
+
+type InMemoryLocker struct {
+	locks map[string][]*Lock
+
+	mx sync.Mutex
```

Comment:
A very widespread convention is for a `sync.Mutex` to be named `mu` and to appear above the definition of struct fields it is intended to protect.